torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8229,12 +8230,13 @@ def _should_disable_fastpath(self, opinfo, rightmost_arg, rightmost_arg_type, dt
         return dtype in integral_types_and(torch.bool) and opinfo.ref in (torch.addcmul,)
 
     def __call__(self, opinfo, device, dtype, requires_grad, **kwargs):
-        num_input_tensors = kwargs.pop("num_input_tensors", foreach_num_tensors)
+        num_input_tensors_specified = "num_input_tensors" in kwargs
+        num_input_tensors = kwargs.pop("num_input_tensors") if num_input_tensors_specified else foreach_num_tensors
         assert isinstance(num_input_tensors, list)
         _foreach_inputs_kwargs = {k: kwargs.pop(k, v) for k, v in _foreach_inputs_default_kwargs.items()}
 
         # zero_size tensor
-        if dtype == torch.float32 and ("cuda" in device):
+        if dtype == torch.float32 and (not num_input_tensors_specified) and ("cuda" in device):
```

Comment:
why are we excluding this case? (just curious)

Reply:
To skip inputs of a bunch of zero-size tensors. See https://github.com/pytorch/pytorch/pull/95028 for the details.